### PR TITLE
docs: fix styles in no-js mode

### PR DESCRIPTION
**Pull request by Tanujkanti4441**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
have fixed the color/styling issues in no-javascript mode.

#### Is there anything you'd like reviewers to focus on?
Fixes: #18076

### CODE REVIEW

File: docs/src/assets/scss/components/logo.scss
Comment (amareshsm) [inline docs/src/assets/scss/components/logo.scss:7]:
```suggestion

```

Do we need to assign a color to the logo title specifically? I don’t recall doing this in the [pr](https://github.com/eslint/eslint/pull/18077/files#diff-06e0f78288fd629ae41ca1ec126819f3f0d4af8d268fd18fc2fb39fbb50c26e6)

Can you check once?

Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/logo.scss:7]: yes, saw that PR! in that logo's title was ok on dark mode but in light mode it's color was like this:

Screenshot 2024-09-17 175000

Comment (amareshsm) [reply docs/src/assets/scss/components/logo.scss:7]: Cool. got it.

**Review by snitin315 — COMMENTED** [review]
Theme & language switcher is not present in no-js mode. Is this expected? Also, does this mean we will always use dark mode in case of no-js mode?

Screenshot 2024-09-19 at 2 27 18 PM

**Comment by amareshsm** [comment]
> Theme & language switcher is not present in no-js mode. Is this expected? Also, does this mean we will always use dark mode in case of no-js mode?
> 
> Screenshot 2024-09-19 at 2 27 18 PM

yes, the theme and language switch won't be there. we will choose a mode based on the system theme. if the system theme is dark we will use dark mode.

**Comment by Tanujkanti4441** [comment]
> Theme & language switcher is not present in no-js mode. Is this expected?

https://github.com/eslint/eslint/blob/8f630eb5794ef9fe38e0b8f034287650def634bd/docs/src/_includes/components/theme-switcher.html#L1

https://github.com/eslint/eslint/blob/8f630eb5794ef9fe38e0b8f034287650def634bd/docs/src/assets/js/themes.js#L22-L23

By default theme switcher component has `hidden` attribute that is removed by js, so i think it's expected that theme switcher is not appearing in no-js mode.

>Also, does this mean we will always use dark mode in case of no-js mode?

yes, that can be an issue, but even if theme-switcher were present it wouldn't be helpful for switching themes in no-js mode, and also all the style retaled issues with no-js mode are in dark theme.

please share if you have any ideas regarding theme switching in no-js mode or should we keep this as is?

File: docs/src/assets/scss/components/alert.scss
Comment (snitin315) [inline docs/src/assets/scss/components/alert.scss:26]: We don't need the border anymore?

File: docs/src/assets/scss/components/alert.scss
Comment (snitin315) [inline docs/src/assets/scss/components/alert.scss:82]: The text `Tip` was slightly bolder earlier.

### after
Screenshot 2024-09-19 at 3 49 31 PM

### before

Screenshot 2024-09-19 at 3 49 54 PM

Comment (snitin315) [reply docs/src/assets/scss/components/alert.scss:82]: https://github.com/user-attachments/assets/d90f4a26-b27f-4c0f-8ce4-a63c9c6f312d

**Comment by snitin315** [comment]
> please share if you have any ideas regarding theme switching in no-js mode or should we keep this as is?

I think it's fine to keep it as is for now, we can come back to this later if required.

File: docs/src/assets/scss/components/alert.scss
Comment (amareshsm) [reply docs/src/assets/scss/components/alert.scss:82]: image

Left image- js disabled - https://deploy-preview-18916--docs-eslint.netlify.app/use/configure/configuration-files
Right image- js enabled - https://eslint.org/docs/latest/use/configure/configuration-files
I don't see any difference in CSS 🤔

File: docs/src/assets/scss/components/alert.scss
Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/alert.scss:26]: Border is already there in parent element (element with `alert` class) and it's color will be the color of alerts accordingly.

File: docs/src/assets/scss/components/alert.scss
Comment (snitin315) [reply docs/src/assets/scss/components/alert.scss:82]: Could it be some global style interference?

**Comment by amareshsm** [comment]
@Tanujkanti4441  Are you still working on this? What’s pending? Could you please resolve the completed conversation?

**Comment by Tanujkanti4441** [comment]
> @Tanujkanti4441 Are you still working on this? What’s pending? Could you please resolve the completed conversation?

Just busy in some academic work, will start working on it soon.

File: docs/src/assets/scss/components/alert.scss
Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/alert.scss:82]: figuring out this, right now `font-weight: 500` seems not be working in no-js mode, but `font-weight: 600` is.

Comment (amareshsm) [reply docs/src/assets/scss/components/alert.scss:82]: Could the change in `font-weight` be due to some fonts not loading properly in no-JS mode?

Comment (amareshsm) [reply docs/src/assets/scss/components/alert.scss:82]: Just a gentle reminder: this is the only item pending. When you’re back, please take a look. Thank you!

**Comment by nzakas** [comment]
Does anyone want to finish this up?

**Comment by amareshsm** [comment]
> Does anyone want to finish this up? 

I'll work on completing this. I've pushed some fixes, and it's nearly finished.

"The text Tip was slightly bolder earlier, same for other alerts too." (ref https://github.com/eslint/eslint/pull/18916#discussion_r1766574830) - I believe the change in font-weight might be caused by some fonts not loading correctly in no-JS mode. This seems to be the only pending issue in this PR."

File: docs/src/assets/scss/components/alert.scss
Comment (amareshsm) [reply docs/src/assets/scss/components/alert.scss:82]: I'm not exactly sure why the font-weight appears slightly lighter in no-JS mode.

**Comment by Tanujkanti4441** [comment]
> > Does anyone want to finish this up?
> 
> I'll work on completing this. I've pushed some fixes, and it's nearly finished.
> 
> "The text Tip was slightly bolder earlier, same for other alerts too." (ref [#18916 (comment)](https://github.com/eslint/eslint/pull/18916#discussion_r1766574830)) - I believe the change in font-weight might be caused by some fonts not loading correctly in no-JS mode. This seems to be the only pending issue in this PR.

Thanks @amareshsm for taking a look, i can also continue to work on this with you.

**Comment by amareshsm** [comment]
> > > Does anyone want to finish this up?
> > 
> > 
> > I'll work on completing this. I've pushed some fixes, and it's nearly finished.
> > "The text Tip was slightly bolder earlier, same for other alerts too." (ref [#18916 (comment)](https://github.com/eslint/eslint/pull/18916#discussion_r1766574830)) - I believe the change in font-weight might be caused by some fonts not loading correctly in no-JS mode. This seems to be the only pending issue in this PR.
> 
> Thanks @amareshsm for taking a look, i can also continue to work on this with you.

if you are back you can take it forward.

File: docs/src/assets/scss/components/alert.scss
Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/alert.scss:82]: https://github.com/eslint/eslint/blob/f16e846ac004bc32e52cd3991d14d7a89374bbb5/docs/src/_includes/layouts/base.html#L108-L115

What i can see here is we are using font with the `regular` font-weight which is not supporting `font-weight:500` in no-js mode, so as a solution we can use the variable font like `Inter-VariableFont_slnt\,wght.ttf` which is working properly.

But i would like to here some opinions on this solutions, may be the current fonts are serving some purpose i am not aware of.

please share if solution seems right.

Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/alert.scss:82]: I tried to fix the font-weight issue of alert's heading can you please check once if this way seems correct? So we can implement the fix with other components also as there are many texts with this issue like in buttons:
Screenshot 2024-10-25 223253

**Comment by amareshsm** [comment]
There are some linting issues that need to be addressed.

File: docs/src/assets/scss/components/alert.scss
Comment (amareshsm) [reply docs/src/assets/scss/components/alert.scss:82]: Sure. I will check.

File: docs/src/assets/scss/components/alert.scss
Comment (amareshsm) [inline docs/src/assets/scss/components/alert.scss:63]:
```suggestion
    font-family: "Inter Variable", sans-serif;
```
Why is the `font-family` explicitly specified as `"Inter Variable", sans-serif`? Is this necessary?

Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/alert.scss:63]: Actually i tried different fonts from repo and this combination of `font-family: "Inter Variable", sans-serif;` and `font-weight: 600;` giving the same result (font and its weight) we have right now, its also about where we are accessing the font through `@font-face`, if we access the `Inter Variable` font through `base.html` file as we are doing right now, then it's not working.

Comment (harish-sethuraman) [reply docs/src/assets/scss/components/alert.scss:63]: You are probably fixing it in the wrong place? https://github.com/eslint/eslint/blob/db0b844a66ee25483f9619d04346de1a2a0d79fa/docs/src/_includes/layouts/base.html#L135-L148 

this is where the script adds inter semibold to the document. May be we might want to look at what this is doing and then move it to scss approach rather than changing font-family of already existing elements.

Comment (harish-sethuraman) [reply docs/src/assets/scss/components/alert.scss:63]: Lets use scss to load this font as we do for other fonts. If we face any issues we can feel free to make changes its fine.

Comment (Tanujkanti4441) [reply docs/src/assets/scss/components/alert.scss]: Done!

**Review by harish-sethuraman — DISMISSED** [review]
LGTM. Will wait for second approve.

File: docs/src/assets/scss/components/alert.scss
Comment (harish-sethuraman) [reply docs/src/assets/scss/components/alert.scss]: thanks!

**Review by amareshsm — DISMISSED** [review]
LGTM. Waiting for code owner approval.

**Review by amareshsm — APPROVED** [review]
LGTM. Waiting for 2nd review.

**Review by harish-sethuraman — APPROVED** [review]
Can someone from @eslint/eslint-tsc help merging this?

**Review by nzakas — APPROVED** [review]
LGTM. Thanks for all the work here.

**Pull request merged**
